api/src/main/java/com/exadel/frs/core/trainservice/filter/SecurityValidationFilter.java

```diff
@@ -86,21 +86,18 @@ public void doFilter(
 
         var apiKey = headersMap.getOrDefault(X_FRS_API_KEY_HEADER, emptyList());
 
-        val key = apiKey.get(0);
-        try {
-            if (key.length() < 36) {
-                throw new IllegalArgumentException("UUID length is incorrect");
-            }
-            UUID.fromString(key);
-        } catch (Exception e) {
-            val objectResponseEntity = handler.handleBadFormatModelKeyException(new BadFormatModelKeyException());
-            buildException(httpResponse, objectResponseEntity);
+        if (!apiKey.isEmpty()) {
```

Comment:
redundant check - we always have an apiKey

Reply:
But if a request doesn't have X_FRS_API_KEY_HEADER, apiKey is empty and apiKey.get(0) has IndexOutOfBoundsException.